dtcli/server_api.py

```diff
@@ -34,6 +34,8 @@ def validate(extension_zip_file, tenant_url, api_token):
             raise dtcliutils.ExtensionValidationError(response.text)
 
 def upload(extension_zip_file, tenant_url, api_token):
+    if tenant_url.endswith("/"):

```

Comment:
The same thing will happen in `validate` method. I'd like to avoid having to fix it for every single method but not sure if there is any common place that we could use

Reply:
We could use a simplified version of what we use in the python api client where there is a client created where we set the base url and then calls to all of the endpoints just use that, where the base_url is an attribute and this cleanup only is done there.

https://github.com/dynatrace-oss/api-client-python/blob/2298dba7306d3f5b2f1dcecb8738c5a0fb5bd267/dynatrace/http_client.py#L37

Reply:
I think we can also register a callback for the [option](https://github.com/dynatrace-oss/dt-cli/blob/2b6bf96f67b30945a077982a898c068ed0a095e7/dtcli/scripts/dt.py#L378 ) and sanitize it therehttps://click.palletsprojects.com/en/8.0.x/advanced/#parameter-modifications

Reply:
That works well, just tried it and pushed a new commit here. Not sure if we should if we have any preference for the method name or where it should go but it works for both validate and upload (just need to remember to include that on any going forward).

If any preferences I can move things around/create a new cleaner PR if we like this approach.

Reply:
Looks great!